backend/pipeline/serializers/crud.py

```diff
@@ -32,10 +62,10 @@ def create(self, validated_data: dict[str, Any]) -> Any:
 
     def save(self, **kwargs: Any) -> Pipeline:
         pipeline: Pipeline = super().save(**kwargs)
-        SchedulerHelper.add_job(
-            str(pipeline.pk),
-            cron_string=pipeline.cron_string,
-        )
+        if pipeline.cron_string is None:
+            SchedulerHelper.remove_job(pipeline_id=str(pipeline.id))
```

Comment:
@muhammad-ali-e since we remove the job when `cron_string` is made `None` - consider the case where we create a pipeline with a valid cron string, run it few times and then remove the cron string. In such a case will we lose information such as `run_count` or the last run time / status?

Reply:
We will only remove the job from Celery's periodic tasks. The last run time and status will still be retained within the Pipeline model.

While we do lose the run_count from Celery, it’s not something we're currently utilizing in our pipeline. If needed, we could store this information in the Pipeline model, but since we can retrieve the count from the workflow_execution table, So this isn't necessary in Pipeline table too.

Additionally, it's important to note that Celery's periodic task table requires at least one scheduling time to function, so if the cron_string is removed, the periodic task must be deleted.